toolkit/tools/pkg/imagecustomizerlib/imagehistory_test.go

```diff
@@ -0,0 +1,179 @@
+package imagecustomizerlib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestConfig(configFilePath string, t *testing.T) imagecustomizerapi.Config {
+	configFile := filepath.Join(testDir, configFilePath)
+
+	var config imagecustomizerapi.Config
+	err := imagecustomizerapi.UnmarshalYamlFile(configFile, &config)
+	assert.NoError(t, err)
+	return config
+}
+
+func TestAddImageHistory(t *testing.T) {
+	tempDir := filepath.Join(tmpDir, "TestAddImageHistory")
+	chroot := safechroot.NewChroot(tempDir, false)
+	err := chroot.Initialize("", []string{}, []*safechroot.MountPoint{}, false)
+	assert.NoError(t, err)
+	defer chroot.Close(false)
+
+	historyDir := filepath.Join(tempDir, customizerLoggingDir)
+	historyFilePath := filepath.Join(historyDir, historyFileName)
+	config := createTestConfig("imagehistory-config.yaml", t)
+	expectedVersion := "0.1.0"
+	expectedDate := time.Now().Format("2006-01-02T15:04:05Z")
+	_, expectedUuid, err := createUuid()
+	assert.NoError(t, err)
+
+	// Test adding the first entry
+	err = addImageHistory(chroot, expectedUuid, testDir, expectedVersion, expectedDate, &config)
+	assert.NoError(t, err, "addImageHistory should not return an error")
+
+	verifyHistoryFile(t, 1, expectedUuid, expectedVersion, expectedDate, config, historyFilePath)
+
+	// Test adding another entry with a different uuid
+	_, expectedUuid, err = createUuid()
+	assert.NoError(t, err)
+	err = addImageHistory(chroot, expectedUuid, testDir, expectedVersion, expectedDate, &config)
+	assert.NoError(t, err, "addImageHistory should not return an error")
+
+	verifyHistoryFile(t, 2, expectedUuid, expectedVersion, expectedDate, config, historyFilePath)
+}
+
+func verifyHistoryFile(t *testing.T, expectedEntries int, expectedUuid string, expectedVersion string, expectedDate string, config imagecustomizerapi.Config, historyFilePath string) {
```

Comment:
Add verification of sshPublicKeys, additionalFiles, etc. to `verifyHistoryFile`?

Reply:
done, removed the test functions for the smaller functions